internal/controller/numaflowcontrollerrollout/numaflowcontrollerrollout_controller.go

```diff
@@ -377,7 +378,19 @@ func (r *NumaflowControllerRolloutReconciler) isControllerDeploymentUpdating(ctx
 	if err != nil {
 		return false, false, err
 	}
-	controllerVersionNeedsToUpdate := (controllerRollout.Spec.Controller.Version != currentVersion)
+
+	numaLogger.Debugf("previousAttemptInstanceID='%s', previousAttemptVersion='%s', currentInstanceID='%s', currentVersion='%s'",
+		controllerRollout.Status.PreviousAttemptStatus.InstanceID, controllerRollout.Status.PreviousAttemptStatus.Version,
+		controllerRollout.Spec.Controller.InstanceID, controllerRollout.Spec.Controller.Version)
+
+	// If the Rollout status indicates previous attempt instanceID and version same to the current instanceID and version,
+	// it means that the Numaplane Controller already attempted to reconcile this instance/version of the Numaflow Controller.
+	// Therefore, independently of the previous attempt result (either success or failure), the reconciler should not perform
+	// the update (but without pausing the pipelines).
+	controllerVersionNeedsToUpdate := (controllerRollout.Spec.Controller.Version != currentVersion &&
+		(controllerRollout.Status.PreviousAttemptStatus.InstanceID != controllerRollout.Spec.Controller.InstanceID ||
+			controllerRollout.Status.PreviousAttemptStatus.Version != controllerRollout.Spec.Controller.Version))
```

Comment:
sorry if I mis-read your slack message. I didn't realize you were still comparing to the version in `getControllerDeploymentVersion()`. I was imagining that we were going to remove that altogether, so the reason I was saying to remove the `DefaultNumaflowControllerImageName` code was because what I really meant was that we should remove the entire `getControllerDeploymentVersion()` function, thus no longer comparing to the live image at all, and only comparing to the previously saved version in the Status. 

Do you see any value with also comparing to the version of the image running?

Reply:
Sorry for lack of clarity